primitives/transaction-pool/src/pool.rs

```diff
@@ -67,16 +67,22 @@ impl PoolStatus {
 /// 2. Inside `Ready` queue:
 ///		- `Broadcast`
 /// 3. Leaving the pool:
-///		- `Finalized`
+///		- `InBlock`
 ///		- `Invalid`
 ///		- `Usurped`
 ///		- `Dropped`
+///	4. Re-entering the pool:
+///		- `Retracted`
+///	5. Block finalized:
+///		- `Finalized`
+///		- `FinalityTimeout`
```

Comment:
What is the timeout?

Reply:
After roughly 512 (a hardcoded constant; currently arbitrary, not based on any research) block pass with no finality we consider that a timeout. This is to prevent prevent leaking memory (and watchers) in case of either:
1. A bug in the code (like missing notification for a particular block)
2. Finality being temporarily broken (or just unavailable)

Note that you are free to re-subscribe for a particular hash after the timeout to wait for even longer (indefinitely)

Reply:
Yeah that is a good idea :) However, I think we should document it. Maybe at least say that `512` is a minimum?

Reply:
Done.